morpheus/_lib/include/morpheus/stages/preprocess_nlp.hpp

```diff
@@ -104,14 +118,230 @@ class PreprocessNLPStage
     int m_stride{-1};
 };
 
+template <typename InputT, typename OutputT>
+PreprocessNLPStage<InputT, OutputT>::PreprocessNLPStage(std::string vocab_hash_file,
+                                                        uint32_t sequence_length,
+                                                        bool truncation,
+                                                        bool do_lower_case,
+                                                        bool add_special_token,
+                                                        int stride,
+                                                        std::string column) :
+  base_t(base_t::op_factory_from_sub_fn(build_operator())),
+  m_vocab_hash_file(std::move(vocab_hash_file)),
+  m_sequence_length(sequence_length),
+  m_truncation(truncation),
+  m_do_lower_case(do_lower_case),
+  m_add_special_token(add_special_token),
+  m_column(std::move(column))
+{
+    // Auto calc stride to be 75% of sequence length
+    if (stride < 0)
+    {
+        stride = m_sequence_length / 2;
+        stride = stride + stride / 2;
+    }
+
+    m_stride = stride;
+}
+
+nvtext::tokenizer_result subword_tokenize(const std::string& vocab_hash_file,
+                                          uint32_t sequence_length,
+                                          bool do_lower_case,
+                                          bool truncation,
+                                          cudf::strings_column_view const& string_col,
+                                          int stride,
+                                          rmm::mr::device_memory_resource* mr);
+
+template <typename InputT, typename OutputT>
+PreprocessNLPStage<InputT, OutputT>::subscribe_fn_t PreprocessNLPStage<InputT, OutputT>::build_operator()
+{
+    return [this](rxcpp::observable<sink_type_t> input, rxcpp::subscriber<source_type_t> output) {
+        return input.subscribe(rxcpp::make_observer<sink_type_t>(
+            [this, &output](sink_type_t x) {
+                auto next = this->pre_process_batch(x);
+
+                output.on_next(std::move(next));
+            },
+            [&](std::exception_ptr error_ptr) {
+                output.on_error(error_ptr);
+            },
+            [&]() {
+                output.on_completed();
+            }));
+    };
```

Comment:
Lets follow how `AddScoresBase` sets up their subscription and just use the `rxcpp::operators::map` function which is equivalent to what is happening here.